x/config/configurator.go

```diff
@@ -41,19 +43,27 @@ type Configurator struct {
 	knownTransports map[string]*compiledTransportSpec
 	knownChoosers   map[string]*compiledChooserSpec
 	knownBinders    map[string]*compiledBinderSpec
+	resolver        interpolate.VariableResolver
 }
 
 // New sets up a new empty Configurator. The returned Configurator does not
 // know about any Transports, peer Chooser lists, or peer list Binders.
 // Individual TransportSpecs, ChooserSpecs, and BinderSpecs must be registered
 // against it using the RegisterTransport, RegisterChooser, and RegisterBinder
 // functions.
-func New() *Configurator {
-	return &Configurator{
+func New(opts ...Option) *Configurator {
+	c := Configurator{
```

Comment:
Instead of taking a pointer each time `c` is used below, maybe just instantiate as a pointer here.